Classes/Handler/FileUploadHandler.php

```diff
@@ -87,6 +123,70 @@ public function uploadFile(InterestRequestInterface $request): ResponseInterface
         );
     }
 
+    /**
+     * @param InterestRequestInterface $request
+     * @return ResponseInterface
+     * @throws \TYPO3\CMS\Extbase\Object\Exception
+     */
+    public function createFileReference(InterestRequestInterface $request): ResponseInterface
```

Comment:
The current implementation of this method is too Product Manager-specific.

If you want to keep this method (and not just use the existing crud API calls to create a sys_file_reference record), you need to make this method work with any scenario. Then I suggest you rewrite this function to handle a file upload like uploadFile, but that you _additionally_ create a file reference and connect the remoteId to that file reference. However, a sys_file_reference can only point to a single file and a single foreign record, so this is a convenience function that must be used with care so the remote side doesn't try to assign it to multiple foreign records.

Reply:
But uploadFile function returns a response object so it responsible for whole response to client :thinking_face: Do we need to separate that part that handling file object and a response in this case?

Reply:
Yes, if you want to handle file upload in two places you need to put that functionality into its own method and call it from fileUpload() and createFileReference() respectively.